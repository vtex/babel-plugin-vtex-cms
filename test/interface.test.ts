import generateInterface from '../src/interfaceGenerator';

const exampleWithAssignments = `
import React from "react";
import RichText from "vtex.store-components/RichText";

const HelloWorld = () => (
  <div>
    <RichText
      textAlignment="CENTER"
      textPosition="CENTER"
      text="# Hello, World!"
    />
  </div>
);

HelloWorld.displayName = 'hello-world';

HelloWorld.schema = {
  title: 'Hello World',
  properties: {
    text: {
      title: "Text",
      type: "string",
    },
    alignment: {
      title: "Alignment",
      type: "string",
    }
  }
}

export default HelloWorld;
`;

describe('Generating an interfaces.json file', () => {
  it('should extract displayName and schema from virtual block', () => {
    const generatedInterface = generateInterface(exampleWithAssignments);

    expect(generatedInterface?.displayName).toEqual('hello-world');
    expect(generatedInterface?.schema).toEqual({
      title: 'Hello World',
      properties: {
        text: {
          title: 'Text',
          type: 'string',
        },
        alignment: {
          title: 'Alignment',
          type: 'string',
        },
      },
    });
  });

  it('should generate interfaces.json file for a virtual block', () => {
    const generatedInterface = generateInterface(exampleWithAssignments);

    expect(generatedInterface?.interfacesFile).toMatchSnapshot();
  });
});
