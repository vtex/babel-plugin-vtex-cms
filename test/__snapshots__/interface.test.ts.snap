// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generating an interfaces.json file should generate interfaces.json file for a virtual block 1`] = `
"{
    \\"hello-world\\": {
      \\"component\\": \\"HelloWorld\\",
      \\"content\\": {\\"title\\":\\"Hello World\\",\\"properties\\":{\\"text\\":{\\"title\\":\\"Text\\",\\"type\\":\\"string\\"},\\"alignment\\":{\\"title\\":\\"Alignment\\",\\"type\\":\\"string\\"}}}
    }
  }"
`;
